app/Controllers/UsersController.php

```diff
@@ -0,0 +1,311 @@
+                } elseif (!password_verify($data['password'], $hashed_password)) {
+                    $data['password_err'] = 'Password does not match';
+                }
+                if (empty($data['email_err']) && empty($data['password_err'])) {
+                    $currentUser = $this->userModel->currentUser($data['email'], $data['password']);
+                    if ($currentUser) {
+                        $this->createUserSession($currentUser);
+                    }
+                }
+                $user = $this->userModel->currentUser($data['email'], $data['password']);
+                if ($user != null) {
+                    if ($user['is_admin'] == 1) {
+                        //if logged in as an admin direct to admin homepage
+                        header('location: ' . 'http://localhost:8000/admin/home');
+                    }
+                    if ($user['is_admin'] == 0) {
+                        //if logged in a general user direct to blogPosts homepage.
+                        header('location: ' . 'http://localhost:8000/blogPosts');
+                    }
+                } else {
+                    return View::make('users/userLogin', $data);
+                }
+            } else {
+                $data = [
+                    'email' => '',
+                    'password' => '',
+                    'email_err' => '',
+                    'password_err' => '',
+                ];
+                return View::make('users/userLogin', $data);
+        }
+    }
+
+    /**
+     *
+     * @return View|void
+     */
+    public function adminHome()
+    {
+        if ($_SESSION != null) {
+            if ($_SESSION['is_admin'] == 1) {
+                $users = $this->userModel->getUserinfo();
+                $data = [
+                    'users' => $users,
+                ];
+                return View::make('/admin/home', $data);
+            } else if ($_SESSION['is_admin'] == 0) {
+                header('location: ' . 'http://localhost:8000/blogPosts');
+            }
+        } else {
+            header('location: ' . 'http://localhost:8000/users/login');
+        }
+    }
+
+    /**
+     * function for logged in admin to create a new user
+     * @return View|void
+     */
+    public function adminAddUser()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == self::REQUEST_METHOD) {
+            $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+            $data = [
+                'name' => trim($_POST['name']),
+                'email' => trim($_POST['email']),
+                'password' => trim($_POST['password']),
+                'confirm_password' => trim($_POST['confirm_password']),
+                'is_admin' => isset($_POST['is_admin']),
+                'name_err' => '',
+                'email_err' => '',
+                'password_err' => '',
+                'confirm_password_err' => ''
+            ];
```

Comment:
This validation below could possibly be used in another function like validateRegistration($data)